providers/hetzner/hetznerProvider.go

```diff
@@ -0,0 +1,168 @@
+package hetzner
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+var features = providers.DocumentationNotes{
+	providers.DocCreateDomains:       providers.Can(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAlias:            providers.Cannot(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseDS:               providers.Cannot(),
+	providers.CanUsePTR:              providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Cannot(),
+	providers.CanUseTLSA:             providers.Cannot(),
+	providers.CanUseTXTMulti:         providers.Cannot(),
+}
+
+func init() {
+	providers.RegisterDomainServiceProviderType("HETZNER", New, features)
+}
+
+// New creates a new API handle.
+func New(settings map[string]string, _ json.RawMessage) (providers.DNSServiceProvider, error) {
+	if settings["api_key"] == "" {
+		return nil, fmt.Errorf("missing HETZNER API Key")
+	}
+
+	api := &api{}
+
+	api.apiKey = settings["api_key"]
+
+	return api, nil
+}
+
+// EnsureDomainExists creates the domain if it does not exist.
+func (api *api) EnsureDomainExists(domain string) error {
+	domains, err := api.ListZones()
+	if err != nil {
+		return err
+	}
+
+	for _, d := range domains {
+		if d == domain {
+			return nil
+		}
+	}
+
+	return api.createZone(domain)
+}
+
+// GetDomainCorrections returns the corrections for a domain.
+func (api *api) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc, err := dc.Copy()
+	if err != nil {
+		return nil, err
+	}
+
+	err = dc.Punycode()
+	if err != nil {
+		return nil, err
+	}
+	domain := dc.Name
+
+	// Get existing records
+	existingRecords, err := api.GetZoneRecords(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	// Normalize
+	models.PostProcessRecords(existingRecords)
+	differ := diff.New(dc)
+	_, create, del, modify, err := differ.IncrementalDiff(existingRecords)
+	if err != nil {
+		return nil, err
+	}
+
+	var corrections []*models.Correction
+
+	zone, err := api.getZone(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range del {
+		record := m.Existing.Original.(*Record)
+		corr := &models.Correction{
+			Msg: fmt.Sprintf("%s, HETZNER ID: %s", m.String(), record.Id),
```

Comment:
Are the HETZNET ID's long?  If that output would look messy, maybe it would be cleaner to either not include it or just include it as `(%s)`